robot/control/Src/motion-control/VelocityKalmanTuner/vkt.py

```diff
@@ -0,0 +1,337 @@
+                    print("Error: Argument provided to -l/--loadfile must be a .txt file!")
+                    self.terminate()
+
+            # Savefile must be a valid.txt file in a valid directory
+            if self.args.savefile:
+                if not os.path.isdir(os.path.split(os.path.abspath(self.args.savefile))[0]):
+                    print("Error: Argument provided to -s/--savefile is not a valid path!")
+                    self.terminate()
+
+                if os.path.splitext(self.args.savefile)[1] != ".txt":
+                    print("Error: Argument provided to -s/--savefile must be a .txt file!")
+                    self.terminate()
+
+            # Outfile must be a valid .cpp file in a valid directory
+            if self.args.outfile:
+                if not os.path.isdir(os.path.split(os.path.abspath(self.args.outfile))[0]):
+                    print("Error: Argument provided to -o/--outfile is not a valid path!")
+                    self.terminate()
+
+                if os.path.splitext(self.args.outfile)[1] != ".hpp":
+                    print("Error: Argument provided to -o/--outfile must be a .cpp file!")
+                    self.terminate()
+
+            if self.args.loadfile:
+                self.load()
+
+            self.simulate()
+
+            if self.args.savefile:
+                self.save()
+
+            if self.args.outfile:
+                self.generate_code()
+
+            self.terminate()
+
+    def terminate(self):
+        self.quit()
+        sys.exit(0)
+
+    def populate_gains(self):
+        for r, row in enumerate(self.kalmanFilter.Q):
+            for c, element in enumerate(row):
+                self.window.Q_textboxes[r][c].setText(str(self.kalmanFilter.Q[r, c]))
+
+        for r, row in enumerate(self.kalmanFilter.R):
+            for c, element in enumerate(row):
+                self.window.R_textboxes[r][c].setText(str(self.kalmanFilter.R[r, c]))
+
+    def update(self):
+        t, x_hat, x = self.kalmanFilter.step(self.dt, self.verbose, changeable_only=True)
+        vx, vy, omega = x_hat
+        vx_obs, vy_obs, omega_obs = x
+        self.vxs.append(vx)
+        self.vys.append(vy)
+        self.omegas.append(omega)
+        self.vxs_obs.append(vx_obs)
+        self.vys_obs.append(vy_obs)
+        self.omegas_obs.append(omega_obs)
+        if self.verbose:
+            self.console_print("-" * 40)
+            v = self.kalmanFilter.vars(True)
+            for key in v:
+                self.console_print(str(key) + ":\n" + str(np.round(v[key], 5)))
+
+    def set_filter_type(self, new_option):
+        self.kalmanFilter.steady_state = (new_option == "Steady State")
+
+    def set_data_type(self, new_option):
+        self.kalmanFilter.step_response = (new_option == "Step Response")
+        if new_option == "Sensor Data":
+            file_name = \
+            QFileDialog.getOpenFileName(self.window.central_widget, "Import Sensor Data", '', "Text File (*.txt)")[0]
```

Comment:
Based on the parsing, it looks to be a csv file, might as well just call it that to be more explicit